### ISSUE TITLE
Alarm time in widget and notification wrong

### ISSUE THREAD

**Issue by th23x** [FossifyOrg/Clock#73]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Clock/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Clock/issues) or [closed](https://github.com/FossifyOrg/Clock/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Clock/discussions) or [closed](https://github.com/FossifyOrg/Clock/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Clock/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.0

### Affected Android/Custom ROM version

Android 12 / Ssmsung stock

### Affected device model

Samsung S10e

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

1. Create an alarm
2. Look at the clock widget / or wait for the upcoming alarm notification - showing the wrong time (see screenshots) 

### Expected behavior

Show alarm target time - in the screenshots 7:20

### Actual behavior

Show 8:25 as next / upcoming alarm

### Screenshots/Screen recordings

<img alt="Screenshot_20240512_071857_Clock" src="https://github.com/FossifyOrg/Clock/assets/13735303/bf9452e7-bec9-4585-8b6b-772c117a5bbc" width=600 />
<img alt="Screenshot_20240512_071852_Clock" src="https://github.com/FossifyOrg/Clock/assets/13735303/01df267c-19e2-4035-b474-26b20832e95e" width=597 />

**Comment by th23x** [comment]
Just to mention... There is this report about alarm snooze behavior with the wrong time ( https://github.com/FossifyOrg/Clock/issues/25 ) but it seems to be a bit different than this one - showing wrong time right away...

**Comment by th23x** [comment]
Sorry, the screenshot of the widget didn't upload properly in my original post - and I don't manage to fix it... 

![Screenshot_20240512_071912_Nova Launcher](https://github.com/FossifyOrg/Clock/assets/13735303/055e72ef-d3b2-4857-acc0-1057f680165b)

**Comment by jaschop-1k5o** [comment]
I am experiencing a similar issue, where the widget is showing the wrong weekday. I have an alarm set to ring every other day, but the widget always shows an upcoming alarm for the next day.

(Context: v1.0.0, Android 13)

**Comment by naveensingh** [comment]
@th23x 

- What timezone are you in? Europe/Berlin?
- Are you able to reproduce this reliably or does it happen randomly?

**Comment by th23x** [comment]
Yes, timezone is Germany / Berlin (GMT +1 currently at winter time)

Indeed, very reproducible, unfortunately

**Comment by naveensingh** [comment]
It would be great if you could test out a fix: 

 - Signed: [clock-test-release.apk.zip](https://github.com/user-attachments/files/19150603/clock-test-release.apk.zip) (APK signed using official keys, compiled locally on my computer)
 - Debug: https://github.com/FossifyOrg/Clock/actions/runs/13749603009/artifacts/2718225070

Both APKs are compiled using https://github.com/FossifyOrg/Clock/pull/99.

**Comment by th23x** [comment]
> It would be great if you could test out a fix

Works, the widget is now showing the correct alarm time 👍

**Comment by naveensingh** [comment]
Thanks!

**Comment by naveensingh** [comment]
Fixed by https://github.com/FossifyOrg/Clock/pull/99

A new update will be available soon.
